### PR TITLE
Dashboard Cards: Activity Card Foundation

### PR DESCRIPTION
**Pull request by zwarm**
Parent #18210 

This PR adds the foundational pieces for the Activity Card dashboard item. To keep this PR small it only includes behind the scenes work, no card is displayed.

Includes
- String and dimen resources
- Activity card layout and card items layout
- ActivityCard class structures (this will be further built out in later PRs)
- View holders and adapters (this will be further built out in later PRs)
- Updates to `CardsAdapter`
- Updates to `MySIteCardAndItem`
- Updates to `CardsTracker` and `CardsShownTracker`

Note: The remainder of the build out will take place in a separate PRs and will include builder and params, building out `ActivityItem`, tweaking the UI and displaying the card.

To test:
There is nothing to test in this PR, as this is all foundational work.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/layout/my_site_activity_card_item.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:clipChildren="false"
+    android:clipToPadding="false"
+    android:foreground="?attr/selectableItemBackground"
+    android:paddingBottom="@dimen/my_site_card_row_padding"
+    android:paddingEnd="@dimen/my_site_card_row_padding"
+    android:paddingStart="@dimen/my_site_card_row_padding"
+    android:paddingTop="@dimen/my_site_card_row_padding">
```
Comment (irfano) [inline WordPress/src/main/res/layout/my_site_activity_card_item.xml:13]: When I need to set four different padding with the same value, I prefer to use `android:padding="@dimen/my_site_card_row_padding"`. Therefore, I suggest using that.

File: WordPress/src/main/res/layout/my_site_activity_card_item.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:background="@drawable/bg_oval_neutral_30"
+        tools:src="@drawable/ic_comment_white_24dp" />
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="@dimen/margin_extra_large"
+        android:layout_marginStart="@dimen/margin_extra_large"
```
Comment (irfano) [inline WordPress/src/main/res/layout/my_site_activity_card_item.xml:32]: np: `marginEnd` and `marginStart` could be combined as `android:layout_marginHorizontal="@dimen/margin_extra_large"`

File: WordPress/src/main/res/layout/my_site_activity_card_item.xml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/activity_card_item_title"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            style="@style/MySitePostItemTitle"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintWidth_default="wrap"
```
Comment (irfano) [inline WordPress/src/main/res/layout/my_site_activity_card_item.xml:46]: Do we have a specific benefit by having `app:layout_constraintWidth_default="wrap"` here and in `activity_card_item_subtitle`? It's not well documented, but it seems `app:layout_constraintWidth_default="wrap"` is [deprecated](https://cs.android.com/androidx/constraintlayout/+/main:constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintLayout.java;l=3583-3585?q=file:androidx%2Fconstraintlayout%2Fwidget%2FConstraintLayout.java%20class:androidx.constraintlayout.widget.ConstraintLayout).

File: WordPress/src/main/res/layout/my_site_activity_card_item.xml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        android:layout_margin="@dimen/margin_extra_large"
+        android:importantForAccessibility="no"
+        android:padding="@dimen/margin_medium"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:background="@drawable/bg_oval_neutral_30"
+        tools:src="@drawable/ic_comment_white_24dp" />
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (irfano) [inline WordPress/src/main/res/layout/my_site_activity_card_item.xml:28]: I'm not sure if we need this inner `ConstraintLayout` here. Maybe this layout will be improved in later PRs. If not I suggest removing the inner `ConstraintLayout` and using a single layout in this file.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt [hunk 5]
```diff
@@ -45,7 +45,7 @@ sealed class MySiteCardAndItem(open val type: Type, open val activeQuickStartIte
         SINGLE_ACTION_CARD,
         JETPACK_FEATURE_CARD,
         JETPACK_SWITCH_CARD,
-        JETPACK_INSTALL_FULL_PLUGIN_CARD
+        JETPACK_INSTALL_FULL_PLUGIN_CARD,
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt:48]: np: I guess you forgot to remove `,`.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt [hunk 6]
```diff
@@ -60,7 +60,8 @@ sealed class MySiteCardAndItem(open val type: Type, open val activeQuickStartIte
         PROMOTE_WITH_BLAZE_CARD,
         DASHBOARD_DOMAIN_CARD,
         PAGES_CARD_ERROR,
-        PAGES_CARD
+        PAGES_CARD,
+        ACTIVITY_CARD,
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt:64]: np: It doesn't break the compiler, but I prefer to not have an extra comma after the last item in the enum class. 🙂

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsTracker.kt [hunk 7]
```diff
@@ -25,8 +25,9 @@ class CardsTracker @Inject constructor(
         POST("post"),
         BLOGGING_PROMPT("blogging_prompt"),
         PROMOTE_WITH_BLAZE("promote_with_blaze"),
+        PAGES("pages"),
+        ACTIVITY("activity"),
         DASHBOARD_CARD_DOMAIN("dashboard_card_domain"),
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsTracker.kt:30]: np: Sorry for my `,` nitpicking. Feel free to ignore them, but another extra comma for the last item here. 😅

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt [hunk 8]
```diff
@@ -275,6 +276,33 @@ sealed class MySiteCardAndItem(open val type: Type, open val activeQuickStartIte
...
+                    ) : ActivityCard(
+                        dashboardCardType = DashboardCardType.ACTIVITY_CARD,
+                        footerLink = footerLink
+                    ) {
+                        data class ActivityItem(
+                            val title: UiString,
+                            val subtitle: UiString?,
+                            @DrawableRes val primaryImage: Int,
+                            val iconColor: Int? = null,
+                            val onClick: ListItemInteraction
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt:296]: We haven't set `primaryImage`, `iconColor` and `iconColor` from view holders. Will we use them in following PRs?

**Review by irfano — APPROVED** [review]
Thank you for being patient with my review. 😄
Great job on the well-structured small PR!
I added some questions, concerns, and nitpicking suggestions. They are all ignorable, so I'm approving the PR but not merging it to give you the flexibility to address my comments.

File: WordPress/src/main/res/layout/my_site_activity_card_item.xml (hunk 1 above)
Comment (zwarm) [reply WordPress/src/main/res/layout/my_site_activity_card_item.xml:13]: Thanks @irfano - this layout serves as the placeholder. Most of this has been cleaned up in later branches.

File: WordPress/src/main/res/layout/my_site_activity_card_item.xml (hunk 2 above)
Comment (zwarm) [reply WordPress/src/main/res/layout/my_site_activity_card_item.xml:32]: Same here @irfano . This layout was a copy/paste and has been updated in later branches.

File: WordPress/src/main/res/layout/my_site_activity_card_item.xml (hunk 3 above)
Comment (zwarm) [reply WordPress/src/main/res/layout/my_site_activity_card_item.xml:46]: Same as above. :)

File: WordPress/src/main/res/layout/my_site_activity_card_item.xml (hunk 4 above)
Comment (zwarm) [reply WordPress/src/main/res/layout/my_site_activity_card_item.xml:28]: This I hadn't considered, but I will apply in a future PR. Thanks for the suggestion!

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt (hunk 5 above)
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt:48]: This started happening after I installed copolit. However I did check that the trailing comma passes checkstyle, lint, and detekt before I left them. Is there really a right way or wrong way or is it just a matter of personal preference? Not something worth falling on my sword for, so I'll jot it down on my list to pull them out later on.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt (hunk 6 above)
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt:64]: Same as above

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsTracker.kt (hunk 7 above)
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsTracker.kt:30]: Same as above.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt (hunk 8 above)
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt:296]: Nope, they are gone. Just placeholder logic here.

**Comment by zwarm** [comment]
@irfano - Thanks so much for the review. I appreciate the time you took to provide the feedback; however I won't be making any changes directly to this PR since I'm already 4 branches ahead locally. I did jot down a few of your nitpicks to address in a future PR. 🙇

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt (hunk 5 above)
Comment (irfano) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt:48]: An extra comma fails in Java but not in Kotlin. There's no warning for it. Therefore, it's mostly a personal style preference, so feel free to ignore it.

**Pull request merged**